src/usersInvites/usersInvites.service.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpException, HttpStatus, Inject, Injectable } from '@nestjs/common';
+import { UserInvite } from '../mail/interface/user.interface';
+import {
+  FITNESS_LOGGER_SERVICE,
+  FitnessLoggerService,
+} from '../logger/logger.service';
+import { MAIL_SERVICE, IMailService } from '../mail/mail.service';
+import { IUsersInvitesDAO, USERS_INVITES_DAO } from '../DAO/usersInvitesDAO';
+import { USERS_SERVICE, IUsersService } from '../users/users.service';
+import { UserInvites } from './interface/userInvites.interface';
+import { INVITES_SERVICE, IInvitesService } from '../invites/invites.service';
+import { inviteConfig } from '../config/mail/invite/invite.config';
+import { Invite } from '../invites/interface/invites.interface';
+
+export const USERS_INVITES_SERVICE = 'USERS INVITES SERVICE';
+
+export interface IUsersInvitesService {
+  createAndSendUserInvite(user: UserInvite): Promise<UserInvites>;
+  sendInvite(user: UserInvite, token: string): Promise<UserInvites>;
+  createInvite(expiresAt: Date): Promise<Invite>;
+  createUserInvite(userInvite: UserInvites): Promise<UserInvites>;
+  updateUserInvite(userInvite: UserInvites): Promise<UserInvites>;
+}
+
+@Injectable()
+export class UsersInvitesService implements IUsersInvitesService {
+  constructor(
+    @Inject(MAIL_SERVICE) private mailService: IMailService,
+    @Inject(USERS_SERVICE) private readonly usersService: IUsersService,
+    @Inject(INVITES_SERVICE) private readonly invitesService: IInvitesService,
+    @Inject(USERS_INVITES_DAO) private usersInvitesDAO: IUsersInvitesDAO,
+    @Inject(FITNESS_LOGGER_SERVICE)
+    private readonly logger: FitnessLoggerService,
+  ) {
+    this.logger.setContext('UsersInvitesService');
+  }
+
+  async createAndSendUserInvite(newUser: UserInvite): Promise<UserInvites> {
+    const user = await this.createUser(newUser);
+
+    const invite = await this.createInvite(inviteConfig.expiresAt);
+
+    await this.createUserInvite({ inviteId: invite.id, userId: user.id });
+
+    return await this.sendInvite(user, invite.invite);
+  }
+
+  async sendInvite(newUser: UserInvite, token: string): Promise<UserInvites> {
+    return await this.mailService.sendMail(newUser, token);
+  }
+
+  async createInvite(expiresAt: Date): Promise<Invite> {
+    return await this.invitesService.saveInvite(expiresAt);
+  }
+
+  async createUser(newUser: UserInvite): Promise<UserInvite> {
```

Comment:
No need to create one more method with the name `createUser` (it's up to UserService to implement that functionality). You can directly use `this.usersService.saveUser()`